### PR TITLE
[SPARK-31790][DOCS] cast(long as timestamp) show different result between Hive and Spark

### PR DESCRIPTION
**Pull request by GuoPhilipse**
### What changes were proposed in this pull request?
add docs for sql migration-guide

### Why are the changes needed?
let user know more about the cast scenarios in which Hive and Spark generate different results

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
no need to test

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@GuoPhilipse, let's file a JIRA although it's doc only, it's user-facing.

**Comment by GuoPhilipse** [comment]
fine, i will create a JIRA for it.

**Comment by GuoPhilipse** [comment]
@HyukjinKwon   cast(ts as long) may change to  cast(long as timestamp) ?

**Comment by HyukjinKwon** [comment]
@GuoPhilipse, feel free to fix the PR title and PR description with examples. It's your PR :-)

**Comment by GuoPhilipse** [comment]
:) have corrected and added examples in JIRA

**Comment by GuoPhilipse** [comment]
@HyukjinKwon  what else did i need to do to close the jira and merge the patch ?

File: docs/sql-migration-guide.md [hunk 1]
```diff
@@ -961,3 +961,4 @@ Below are the scenarios in which Hive and Spark generate different results:
 * `SQRT(n)` If n < 0, Hive returns null, Spark SQL returns NaN.
 * `ACOS(n)` If n < -1 or n > 1, Hive returns null, Spark SQL returns NaN.
 * `ASIN(n)` If n < -1 or n > 1, Hive returns null, Spark SQL returns NaN.
+* `CAST(n as timestamp)` If n is Byte/Short/Int/Long data type, Hive treat n as milliseconds unit , while Spark SQL as seconds unit.
```
Comment (HyukjinKwon) [inline docs/sql-migration-guide.md:964]: Let's rephrase a bit more:

`CAST(n AS TIMESTAMP)` If n is integral numbers, Hive treats n as milliseconds, Spark SQL treats n as seconds .

**Comment by HyukjinKwon** [comment]
ok to test

File: docs/sql-migration-guide.md (hunk 1 above)
Comment (GuoPhilipse) [reply docs/sql-migration-guide.md]: cool , have improved it.

**Comment by HyukjinKwon** [comment]
Merged to master

**Pull request closed without merging**